### PR TITLE
Display occupants avatar

### PR DESCRIPTION
**Pull request by raphaelSeguin**
Hi,
this is a first contribution. There are some 
I copied the AvatarMixin from chatbox-views, would have it been better to import the whole file, or maybe abstract this mixin in separate file ?

### CODE REVIEW

File: src/converse-rosterview.js [hunk 1]
```diff
@@ -480,7 +506,11 @@ converse.plugins.add('converse-rosterview', {
                         'desc_chat': __('Click to chat with %1$s (JID: %2$s)', display_name, item.get('jid')),
                         'desc_remove': __('Click to remove %1$s as a contact', display_name),
                         'allow_contact_removal': _converse.allow_contact_removal,
-                        'num_unread': item.get('num_unread') || 0
+                        'num_unread': item.get('num_unread') || 0,
+                        classes: '',
+                        width: '24',
+                        height: '24',
```
Comment (jcbrand) [inline src/converse-rosterview.js]: Looks like the `width` and `height` values you're passing in here never get used.

File: src/converse-rosterview.js [hunk 2]
```diff
@@ -22,6 +23,26 @@ import tpl_search_contact from "templates/search_contact.html";
...
+                image = this.model.vcard.get('image');
+
+        canvas_el.outerHTML = tpl_avatar({
+            'classes': canvas_el.getAttribute('class'),
+            'width': canvas_el.width,
+            'height': canvas_el.height,
+            'image': "data:" + image_type + ";base64," + image,
+        });
+    },
+};
```
Comment (jcbrand) [inline src/converse-rosterview.js]: There's no need to duplicate the code. Instead, in `src/converse-chatboxviews.js`, you can set `AvatarMixin` as an attribute on the `_converse` object.

Then further down you can do:

```_converse.ViewWithAvatar = Backbone.NativeView.extend(_converse.AvatarMixin);```

File: sass/_roster.scss [hunk 3]
```diff
@@ -119,7 +119,9 @@
 
                     .avatar {
                         float: left;
-                        display: inline-block;
+                        //display: inline-block;
+                        margin-right: 0.5em;
+                        //margin-top: -4px;
```
Comment (jcbrand) [inline sass/_roster.scss]: Please don't leave uncommented code anywhere.

File: sass/_roster.scss [hunk 4]
```diff
@@ -119,7 +119,9 @@
 
                     .avatar {
                         float: left;
```
Comment (jcbrand) [inline sass/_roster.scss]: We're using flexbox for many things now, so this `float: left` statement might no longer be necessary.

File: sass/_roster.scss (hunk 3 above)
Comment (raphaelSeguin) [reply sass/_roster.scss]: Oups ! OK !

File: src/converse-rosterview.js (hunk 1 above)
Comment (raphaelSeguin) [reply src/converse-rosterview.js]: I'll remove them then.

File: sass/_roster.scss (hunk 4 above)
Comment (raphaelSeguin) [reply sass/_roster.scss]: Yes. In fact, I found it here, and when I tried to remove it, it broke the design. I'll try to remove it again and fix the display with flexbox.

File: sass/_roster.scss (hunk 3 above)
Comment (raphaelSeguin) [reply sass/_roster.scss]: removed

**Comment by linkmauve** [comment]
This fixes #1322, btw.

**Comment by jcbrand** [comment]
I've made some updates to this PR and then did a manual merge.

**Pull request closed without merging**